File: diapp/app/src/main/java/com/example/diapp/MainActivity.java
package com.example.diapp;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.example.diapp.di.Engine;
import com.example.diapp.di.Plane;
import com.example.diapp.di.PlaneType;
import com.example.diapp.di.Wings;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        Engine e = new Engine();
        Wings w = new Wings();
        PlaneType t = new PlaneType();

        Plane p = new Plane(e, w, t);
        p.TakeOff();

    }
}

File: KotlinCoroutineApp2/app/src/main/java/com/example/kotlincoroutineapp2/MainActivity.kt
package com.example.kotlincoroutineapp2

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.databinding.DataBindingUtil
import com.example.kotlincoroutineapp2.databinding.ActivityMainBinding
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch


class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        SayHelloFromMainThread()

        SayHelloFromBackThread()

    }

    private fun SayHelloFromMainThread() {

        CoroutineScope(Dispatchers.Main).launch {
            binding.text1.text = "Hello From " +
                    " ${Thread.currentThread().name}"
        }
    }

    private fun SayHelloFromBackThread() {
        CoroutineScope(Dispatchers.IO).launch {
            binding.text2.text = " Hello From " +
                    " ${Thread.currentThread().name}"
        }
    }


}



